### PR TITLE
Initial setup for graphql emitter

### PR DESCRIPTION
**Pull request by swatkatz**
Add an initial commit for the emitter

Apologies for re-doing it, my previous branch got polluted because I auto merged some suggestions and tried to pull the commit incorrectly. I've incorporated all the changes in this new commit though

Test:

Ran `npm run watch` to see everything compile without issues
Ran `pnpm run test` to see the hello world test run

### CODE REVIEW

File: packages/graphql/.vscode/extensions.json
Comment (timotheeguerin) [inline packages/graphql/.vscode/extensions.json:1]: this one too is a duplicate to the root one

File: packages/graphql/prettierrc.yaml
Comment (timotheeguerin) [inline packages/graphql/prettierrc.yaml:1]: this one too

File: packages/graphql/src/lib.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { createTypeSpecLibrary } from "@typespec/compiler";
+
+export const $lib = createTypeSpecLibrary({
+  name: "graphql",
```
Comment (timotheeguerin) [inline packages/graphql/src/lib.ts:4]:
```suggestion
  name: "@typespec/graphql",
```

File: packages/graphql/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "extends": "../../tsconfig.base.json",
+  "compilerOptions": {
+    "useDefineForClassFields": true,
+    "rootDir": ".",
+    "outDir": "dist"
```
[the remark this answers is not in the document]
Comment (swatkatz) [reply packages/graphql/tsconfig.json:6]: That's a good suggestion, I'll incorporate it.

File: packages/graphql/.vscode/extensions.json
Comment (swatkatz) [reply packages/graphql/.vscode/extensions.json:1]: done

File: packages/graphql/prettierrc.yaml
Comment (swatkatz) [reply packages/graphql/prettierrc.yaml:1]: done

File: packages/graphql/tsconfig.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (swatkatz) [reply packages/graphql/tsconfig.json]: done

File: packages/graphql/src/lib.ts (hunk 1 above)
Comment (swatkatz) [reply packages/graphql/src/lib.ts]: done

**Review by timotheeguerin — APPROVED** [review]
Looks good! Before or as you start adding the library content(decorators) you might want to also configure `tspd` (internal tool for now) that will generate the decorator signature for you (and can do for docs too later)

```
  "gen-extern-signature": "tspd --enable-experimental gen-extern-signature .",
```

**Pull request merged**